### PR TITLE
Bring latest tests to master to the CI environment will use them.

### PR DESCRIPTION
**Pull request by selimnairb**
Adds tests for:
- OAuth
- Ensuring resource types returned by /hsapi/resourceList correspond to actual resource types (see https://github.com/hydroshare/hydroshare/issues/752)

### CODE REVIEW

**Comment by selimnairb** [comment]
@pkdash Can you review and merge this?

File: tests/api/test_live.py [hunk 1]
```diff
@@ -170,11 +189,50 @@ def get_hs_auth(self):
...
+            if self.port:
+                token_url = self._TOKEN_URL_PROTO_WITH_PORT.format(scheme=scheme,
+                                                                   hostname=self.url,
+                                                                   port=self.port)
+            else:
+                token_url = self._TOKEN_URL_PROTO_WITHOUT_PORT.format(scheme=scheme,
+                                                                      hostname=self.url)
+            r = requests.post(token_url, data=data, auth=(self.client_id, self.client_secret),
+                              verify=self.verify)
+            token = json.loads(r.text)
```
Comment (pkdash) [inline tests/api/test_live.py:226]: I think `token=r.json()`  should work

File: tests/api/test_live.py [hunk 2]
```diff
@@ -298,6 +356,74 @@ def test_netcdf_resource_create(self):
...
+        public_count = 0
+        private_count = 0
+        res_list = hs.getResourceList(creator=self.creator)
+        for res in res_list:
+            if res['public'] is True:
+                public_count += 1
+            else:
+                private_count += 1
+
+        self.assertGreater(private_count,public_count, "Private("+str(private_count)+") not greater than public("+str(public_count)+") ")
```
Comment (pkdash) [inline tests/api/test_live.py:395]: Not sure why we need to test that `private_count` > `public_count` rather just testing   `private_count` is 2 and `public_count` is 1

File: tests/api/test_live.py [hunk 3]
```diff
@@ -298,6 +356,74 @@ def test_netcdf_resource_create(self):
...
```
Comment (pkdash) [inline tests/api/test_live.py:424]: As before

**Pull request merged**